pandas/indexes/multi.py

```diff
@@ -1907,6 +1907,7 @@ def convert_indexer(start, stop, step, indexer=indexer, labels=labels):
                 return np.array(labels == loc, dtype=bool)
             else:
                 # sorted, so can return slice object -> view
+                loc = labels.dtype.type(loc)
```

Comment:
Yeah, I am not sure how much checking would be needed here.

My understanding is that it *probably* is not needed in this case, as `loc` is coming from `loc = level_index.get_loc(key)`, and `labels` and `level_index` are from the same MultiIndex. So I would assume that `get_loc` can only return an existing label, and so should fit in the dtype of `labels`?

(but probably also not that a perf issue to do the check)